classes/task/merge_user_accounts.php

```diff
@@ -0,0 +1,194 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+namespace tool_mergeusers\task;
+use MergeUserTool;
+use \tool_mergeusers\merge_request;
+/**
+ * Adhoc task to merge user accounts.
+ *
+ * @package   tool_mergeusers
+ * @author    Jordi Pujol-Ahulló <jordi.pujol@urv.cat>
+ * @author    Nicola Vallinoto <n.vallinoto@liguriadigitale.it>
+ * @copyright 2013 Servei de Recursos Educatius (http://www.sre.urv.cat)
+ * @copyright 2023 Liguria Digitale (http://www.liguriadigitale.it)
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * */
+class merge_user_accounts extends \core\task\adhoc_task {
+    /**
+     * Execute the task of merge users accounts.
+     */
+    public function execute() {
```

Comment:
As with the other task, it would be interesting to use `mtrace()` to report what was found by this task. And for instance, report the userfield + uservalue for both user to keep and user to remove. To provide a context into the task log into Moodle.

Reply:
ok

Reply:
As with the other task, do NOT mtrace the start and end of the execution of the adhoc task, just useful information we need just in case we have some troubles and need to follow its execution.

Thinking about privacy, I think we should not mtrace the pairs userfield + uservalue, but just the merge request id. The merge request id will be kept in the database model, so we always will be able to follow which users were merged.

Reply:
I removed mtrace at the beginning and ending of execute function of adhoc task.